user_external/lib/imap.php

```diff
@@ -49,6 +49,32 @@ public function checkPassword($uid, $password) {
 			imap_close($mbox);
 			$uid = mb_strtolower($uid);
 			$this->storeUser($uid);
+			
+			if (preg_match('/^([^@]+)@(.+)\.[^.]+$/', $uid, $matches)) { 
+			  // uid is formatted username@domain.extension
+			  $username=$matches[1];
+			  $username=str_replace(".", " ", $username);
+			  $domain=$matches[2];
+			  
+			  // set email adress, if not already defined
+			  $userManager=\OC::$server->getUserManager();
+			  $user=$userManager->get($uid);
+			  $currentEmail = $user->getEMailAddress();
+			  if ($currentEmail == "") {
+			    $user->setEMailAddress($uid);
+			  }
+
+			  // set Display name, if not already defined
+			  $displayname=$this->getDisplayName($uid) ;
+			  if ($displayname == "") {
+			    $this->setDisplayName($uid, $username) ;
+			  }
+
+			  // create group from domain name, add user to group
+			  OC_Group::createGroup($domain);
+			  OC_Group::addToGroup($uid, $domain);
```

Comment:
Just checked, it seems that these will fail silentely in case the group or membership already exists, which is ok for your use case.